src/ScaledDomains.Extensions.Caching.MySql/MySqlServerCachingServicesExtensions.cs

```diff
@@ -27,13 +30,34 @@ public static IServiceCollection AddDistributedMySqlServerCache(this IServiceCol
                 throw new ArgumentNullException(nameof(setupAction));
             }
 
-            services.AddOptions();
-
+            services.TryAddEnumerable(ServiceDescriptor.Singleton<IValidateOptions<MySqlServerCacheOptions>, ValidateMySqlServerCacheOptions>());
+            services.Configure(setupAction);
+            services.Add(ServiceDescriptor.Singleton<ISystemClock, SystemClock>());
```

Comment:
I think in this case would be better to use `TryAdd` method or should introduce custom interface (eg: IMySqlServerCacheClock) to avoid multiple registration.

Reply:
Decided to add custom implementation. It was not a good idea to reuse a Microsoft internal interface anyway.